leaf-ui/js/onFunctions.js

```diff
@@ -1034,23 +1003,83 @@ paper.on('blank:pointerdown', function(evt, x, y) {
 });
 
 /**
- *
+ * Specifies behavior for clicking on cells and moving intentions/links
  */
-paper.on('cell:pointerdown', function(cellView, evt, x, y) {
-	
-	if(mode == "Analysis"){
-		return;
-	}
-
-	var cell = cellView.model;
-	if (cell instanceof joint.dia.Link){
-		cell.reparent();
-	}
+paper.on({
+    'cell:pointerdown': function(cellView, evt) {
+        var interact = true;
+        if(mode == "Analysis"){
+            interact = false;
+        }
 
-	// Unembed cell so you can move it out of actor
-	if (cell.get('parent') && !(cell instanceof joint.dia.Link)) {
-		graph.getCell(cell.get('parent')).unembed(cell);
-	}
+        // pass data to pointermove and pointerup
+        evt.data = {move: false, interact: interact};
+    },
+    'cell:pointermove': function(cellView, evt) {
+        if (!evt.data.move && evt.data.interact){
+            // start of a click and drag
+            evt.data.move = true;
+        }
+      },
+    'cell:pointerup': function(cellView, evt) {
+        // undefined event occurs due to paper.trigger("cell:pointerup"...
+        // on adding new cell to paper
+        if (evt == undefined) {
```

Comment:
evt is undefined when we add a new cell to the paper, because we call paper.trigger("cell:pointerup" without passing in evt data from a pointerdown or pointermove. All the cell highlighting, inspector appearing, re-embedding  in actor, etc. that happens in pointerup should occur when a new cell is added, so this defines evt.data such that those will occur. (Adjusted this code since Ali's review earlier today—I was skipping undefined evts and calling an extra embedBasicActor when adding intentions, but it didn't preserve the highlighting and inspector behavior.)